src/NServiceBus.Core/Transports/Learning/AsyncDirectory.cs

```diff
@@ -0,0 +1,31 @@
+namespace NServiceBus
+{
+    using System.IO;
+    using System.Threading.Tasks;
+
+    static class AsyncDirectory
+    {
+        public static async Task Move(string sourcePath, string targetPath)
+        {
+            var count = 0;
+            while (count <= 10)
```

Comment:
This does mean that we'll be trying 11 times, is that what you intended, or should this be `count < 10`?

Reply:
The actual number is arbitrary as long as we throw after the last retry failed. Feel free to change it to your liking